qa/tasks/deepsea.py

```diff
@@ -720,6 +723,52 @@ def _run_orch(self, orch_tuple):
                 cmd_str += ' 2>/dev/null'
         if self.dev_env:
             cmd_str = 'DEV_ENV=true ' + cmd_str
+        self._handle_execution(cmd_str, orch_spec, survive_reboot=survive_reboot)
+
+    def _handle_execution(self, cmd_str, orch_spec, survive_reboot=False):
+        """
+        TODO: write docstring
+        """
+        if survive_reboot:
+            self._exec_with_retry(cmd_str, orch_spec)
+        else:
+            self._exec(cmd_str, orch_spec)
+
+    def _exec_with_retry(self, cmd_str, orch_spec):
```

Comment:
I would suggest refactor _exec method to support named argument `tries` defaults 1 instead of creating separate method.

Reply:
The difference between `_exec_with_retry` and `_exec` is mainly the `ConnectionLostError` exception handling. As you can't construct a conditional in a try-except block, the refactoring to a single method would be a concatenation of two try-except blocks. Imo this looks bloated and doesn't read very nicely.

Reply:
Not sure I understand why it is bloated? My points:
1) Having two methods introduce code duplication
2) The methods actually do the same job especially one can use _exec_with_retry with max_attempts=1 without changes with the same result as _exec.

Reply:
@jschmid1 so, do I understand correctly that 
1) you want retry only if ConnectionLostError exception occurred?
2) you don't want catch ConnectionLostError in regular case.

Reply:
Nope, I **only** want to check for ConnectionLostError when there is a reboot exepcted (check '_detect_reboot()')